pkg/metrics/engine.go

```diff
@@ -8,103 +8,197 @@ import (
-		existsDuration                prometheus.Counter
-		getDuration                   prometheus.Counter
-		headDuration                  prometheus.Counter
-		inhumeDuration                prometheus.Counter
-		putDuration                   prometheus.Counter
-		rangeDuration                 prometheus.Counter
-		searchDuration                prometheus.Counter
-		listObjectsDuration           prometheus.Counter
-		containerSize                 prometheus.GaugeVec
-		payloadSize                   prometheus.GaugeVec
+		listContainersDuration_counter        prometheus.Counter
+		estimateContainerSizeDuration_counter prometheus.Counter
+		deleteDuration_counter                prometheus.Counter
+		existsDuration_counter                prometheus.Counter
+		getDuration_counter                   prometheus.Counter
+		headDuration_counter                  prometheus.Counter
+		inhumeDuration_counter                prometheus.Counter
+		putDuration_counter                   prometheus.Counter
+		rangeDuration_counter                 prometheus.Counter
+		searchDuration_counter                prometheus.Counter
+		listObjectsDuration_counter           prometheus.Counter
+
+		listContainersDuration        prometheus.Histogram
+		estimateContainerSizeDuration prometheus.Histogram
+		deleteDuration                prometheus.Histogram
+		existsDuration                prometheus.Histogram
+		getDuration                   prometheus.Histogram
+		headDuration                  prometheus.Histogram
+		inhumeDuration                prometheus.Histogram
+		putDuration                   prometheus.Histogram
+		rangeDuration                 prometheus.Histogram
+		searchDuration                prometheus.Histogram
+		listObjectsDuration           prometheus.Histogram
+
+		containerSize prometheus.GaugeVec
+		payloadSize   prometheus.GaugeVec
 	}
 )
 
 const engineSubsystem = "engine"
 
 func newEngineMetrics() engineMetrics {
 	var (
-		listContainersDuration = prometheus.NewCounter(prometheus.CounterOpts{
+		listContainersDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+			Namespace: storageNodeNameSpace,
+			Subsystem: engineSubsystem,
+			Name:      "list_containers_time",
+			Help:      "Engine list containers handling time",
+		})
+
+		estimateContainerSizeDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+			Namespace: storageNodeNameSpace,
+			Subsystem: engineSubsystem,
+			Name:      "estimate_container_size_time",
+			Help:      "Engine container size handling time",
+		})
+
+		deleteDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+			Namespace: storageNodeNameSpace,
+			Subsystem: engineSubsystem,
+			Name:      "delete_time",
+			Help:      "Engine delete operations handling time",
+		})
+
+		existsDuration = prometheus.NewHistogram(prometheus.HistogramOpts{
+			Namespace: storageNodeNameSpace,
+			Subsystem: engineSubsystem,
+			Name:      "exists_time",
+			Help:      "Engine exists operations handling time",
```

Comment:
`Engine "exists" operation...`?

Reply:
do you mean quoting operation names?

Reply:
Yeah, I'm not sure it's clear what is meant this way.

Reply:
fixed (single quote)